### PR TITLE
Only output log files if tests fail

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

**Comment by pwittrock** [comment]
This is to make it easier to debug flaky tests.

File: test/e2e_node/e2e_node_suite_test.go [hunk 1]
```diff
@@ -78,15 +81,39 @@ var _ = AfterSuite(func() {
...
-		b.WriteString(fmt.Sprintf("etcd output:\n%s", e2es.etcdCombinedOut.String()))
+		b.WriteString(fmt.Sprintf("apiserver output:\n%s\n", e2es.apiServerCombinedOut.String()))
 		b.WriteString("-------------------------------------------------------------\n")
-		glog.V(2).Infof(b.String())
-
+		b.WriteString(fmt.Sprintf("etcd output:\n%s\n", e2es.etcdCombinedOut.String()))
 	}
-})
+	b.WriteString("******************************************************\n")
+	glog.V(2).Infof(b.String())
```
Comment (vishh) [inline test/e2e_node/e2e_node_suite_test.go]: Why aren't we logging by default?

Comment (pwittrock) [inline test/e2e_node/e2e_node_suite_test.go]: Because it makes the output hard to parse.  If the test suite is successful, I don't think we want to output all the logs.

Comment (vishh) [inline test/e2e_node/e2e_node_suite_test.go]: This PR ensures that we do not print anything unless the test failed right? So other than an empty line, how else will it affect a successful test run?

Comment (pwittrock) [inline test/e2e_node/e2e_node_suite_test.go]: Things are particularly annoying when you have tests run against 6 different hosts and you want to look at the kubelet.logs for the failed host.

Comment (vishh) [inline test/e2e_node/e2e_node_suite_test.go]: Sorry if I was not clear. The rationale to print logs only on failure is very sound. But why `V(2)` instead of `V(0)`?

Comment (pwittrock) [inline test/e2e_node/e2e_node_suite_test.go]: No strong reason.  I will update this in a follow up.
